plugins/inputs/ctrlx_datalayer_sse/ctrlx_datalayer_sse_test.go

```diff
@@ -0,0 +1,241 @@
+		require.NoError(t, err)
+	}))
+	// Handle request to validate token
+	mux.HandleFunc("/identity-manager/api/v2/auth/token/validity", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, err := w.Write([]byte("{\"valid\": \"true\"}"))
+		require.NoError(t, err)
+	}))
+	// Handle request to create subscription
+	mux.HandleFunc(path, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusCreated)
+		_, err := w.Write([]byte("201 created"))
+		require.NoError(t, err)
+	}))
+	// Handle request to fetch sse data
+	mux.HandleFunc(path+"/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == http.MethodGet {
+			w.WriteHeader(http.StatusOK)
+			_, err := w.Write([]byte("event: update\n"))
+			require.NoError(t, err)
+			_, err = w.Write([]byte("id: 12345\n"))
+			require.NoError(t, err)
+			if getMultiEntries() {
+				data := "data: {\n"
+				_, err = w.Write([]byte(data))
+				require.NoError(t, err)
+				data = "data: \"node\":\"plc/app/Application/sym/PLC_PRG/counter\", \"timestamp\":132669450604571037,\"type\":\"double\",\"value\":44.0\n"
+				_, err = w.Write([]byte(data))
+				require.NoError(t, err)
+				data = "data: }\n"
+				_, err = w.Write([]byte(data))
+				require.NoError(t, err)
+			} else {
+				data := "data: {\"node\":\"plc/app/Application/sym/PLC_PRG/counter\", \"timestamp\":132669450604571037,\"type\":\"double\",\"value\":43.0}\n"
+				_, err = w.Write([]byte(data))
+				require.NoError(t, err)
+			}
+			_, err = w.Write([]byte("\n"))
+			require.NoError(t, err)
+		}
+	}))
+	return httptest.NewServer(mux)
+}
+
+func cleanup(server *httptest.Server) {
+	server.CloseClientConnections()
+	server.Close()
+}
+
+func initRunner(t *testing.T) (*CtrlXDataLayerSSE, *httptest.Server) {
+	server := newServer(t)
+
+	subs := make([]Subscription, 0)
+	subs = append(subs, Subscription{
+		Measurement: "ctrlx",
+		Nodes: []Node{
+			{FieldName: "counter", Address: "plc/app/Application/sym/PLC_PRG/counter"},
+		},
+	},
+	)
+	s := &CtrlXDataLayerSSE{
+		connection:         &http.Client{},
+		url:                server.URL,
+		Username:           "user",
+		Password:           "password",
+		InsecureSkipVerify: true,
+		Subscription:       subs,
+		tokenManager:       token.TokenManager{Url: server.URL, Username: "user", Password: "password", Connection: &http.Client{}},
+		Log:                testutil.Logger{},
+	}
+	return s, server
+}
+
+func TestCtrlXMetricsField(t *testing.T) {
+	const measurement = "ctrlx"
+	const fieldName = "counter"
+	s, server := initRunner(t)
+	defer cleanup(server)
+	var acc testutil.Accumulator
+	time.Sleep(time.Microsecond * 1000)
+	require.NoError(t, acc.GatherError(s.Start))
+	time.Sleep(time.Millisecond * 2000)
+	v, found := acc.FloatField(measurement, fieldName)
+	require.True(t, found, "expected value to be a float")
+	require.Equalf(t, 43.0, v, "expected value to be 43, but got %v", v)
```

Comment:
Can we add some newlines to carve-out functional blocks and make this more readable!?!? Same for the instances below.